driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -137,6 +138,18 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             : RetryDecision.rethrow();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onRequestError(Statement statement, ConsistencyLevel cl, DriverException e, int nbRetry) {
```

Comment:
The implementation here simply mimics the legacy behavior, which was to retry on next host. Changing this to rethrow would be a significant behavioral change as users would start getting exceptions where previously the operation would be retried; so I'm not sure if we should change it, but I'm open to discussion. BTW in FallthroughRetryPolicy I did exactly that (i.e. it retries on next host in 2.1, but rethrows in 3.0).